src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -72,8 +72,8 @@ export const useSwapper = () => {
   const { setValue, clearErrors, getValues } = useFormContext()
   const translate = useTranslate()
   const isComponentMounted = useIsComponentMounted()
-  const [quote, trade, sellAsset] = useWatch({
-    name: ['quote', 'trade', 'sellAsset'],
+  const [quote, trade, sellAsset, buyAsset] = useWatch({
+    name: ['quote', 'trade', 'sellAsset', 'buyAsset'],
   })
```

Comment:
```suggestion
  const [quote, trade, sellAsset, buyAsset] = useWatch({
    name: ['quote', 'trade', 'sellAsset', 'buyAsset'],
  }) as [Quote, Trade, Asset, Asset]
```

`useWatch` is unable to unpack nested types, we might want to cast this to something more useful, as i believe it returns `any[]`

alternatively, if we break it up to n separate `useWatch` lines rather than passing an array, it's able to correctly type them IIRC

Reply:
Indeed it does return `any[]` - we should move away from nested types with `useWatch` as it can cause us to miss bugs like the optional chaining issue yesterday.